src/firefly/java/edu/caltech/ipac/firefly/server/dpanalyze/SofiaAnalyzer.java

```diff
@@ -0,0 +1,285 @@
+package edu.caltech.ipac.firefly.server.dpanalyze;
+
+import edu.caltech.ipac.firefly.core.FileAnalysisReport;
+import edu.caltech.ipac.firefly.data.sofia.Sofia1DSpectraExtractor;
+import edu.caltech.ipac.firefly.data.sofia.SofiaSpectraModel;
+import edu.caltech.ipac.firefly.data.sofia.VOSpectraModel;
+import edu.caltech.ipac.firefly.server.ServerContext;
+import edu.caltech.ipac.table.DataGroup;
+import edu.caltech.ipac.table.io.VoTableReader;
+import edu.caltech.ipac.table.io.VoTableWriter;
+import edu.caltech.ipac.util.download.FailedRequestException;
+import edu.caltech.ipac.util.download.URLDownload;
+import nom.tam.fits.Header;
+
+import java.io.File;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.util.*;
+
+import static edu.caltech.ipac.table.TableUtil.Format.VO_TABLE_TABLEDATA;
+
+/*
+ Analyzer for SOFIA extracted Spectra from FITS 'image'
+ */
+@DataProductAnalyzerImpl(id = "analyze-sofia")
+public class SofiaAnalyzer implements DataProductAnalyzer {
+
+
+    @Override
+    public FileAnalysisReport analyze(FileAnalysisReport inputReport, File inFile, String analyzerId, Map<String, String> params) {
+        return inputReport;
+    }
+
+    @Override
+    public FileAnalysisReport analyzeFits(FileAnalysisReport inputReport,
+                                          File inFile,
+                                          String analyzerId,
+                                          Map<String, String> params,
+                                          Header[] headerAry) {
+
+
+        String code = params.getOrDefault("product_type", "");
+        String level = params.getOrDefault("processing_level", "");
+        String inst = params.getOrDefault("instrument", "");
```

Comment:
It might be easier to go ahead and convert all to uppper case here. It would make the `if`'s look a little cleaner since you could just use `equals`

```java
String inst = params.getOrDefault("instrument", "").toUpperCase();\
.
.
.
if (inst.equals("FORCAST")) {
```